### PR TITLE
feat(server): add JSON export for schema

### PR DESCRIPTION
**Pull request by nourbalaha**
# Overview
This PR introduces a feature for exporting schemas in JSON format.

## Release Notes

- **New Features**
  - Introduced multiple new API endpoints for exporting schemas in JSON format, enhancing data retrieval capabilities.
  - Added integration tests for schema export functionality, ensuring robust API behavior under various conditions.

- **Bug Fixes**
  - Improved error handling for unauthorized access and non-existent schemas in the API.

- **Documentation**
  - Updated OpenAPI specification to include new endpoints and response structures, ensuring clarity for API consumers.

- **Tests**
  - Added unit tests for schema management functions to validate expected behavior and outputs.

### CODE REVIEW

File: server/internal/adapter/publicapi/types.go [hunk 1]
```diff
@@ -212,6 +213,28 @@ func NewItemAsset(a *asset.Asset, urlResolver asset.URLResolver) ItemAsset {
...
 
+const defaultJSONSchemaVersion = "https://json-schema.org/draft/2020-12/schema"
+
+type SchemaJSON struct {
+	Schema      *string                 `json:"schema,omitempty"`
+	Id          *string                 `json:"id,omitempty"`
+	Title       *string                 `json:"title,omitempty"`
+	Description *string                 `json:"description,omitempty"`
+	Type        *string                 `json:"type,omitempty"`
+	Properties  *map[string]interface{} `json:"properties,omitempty"`
```
Comment (yk-eukarya) [inline server/internal/adapter/publicapi/types.go:224]: why nullable? if not on purpose you can change it from the yml file

File: server/internal/usecase/interactor/schema.go [hunk 2]
```diff
@@ -72,6 +72,18 @@ func (i Schema) FindByModel(ctx context.Context, mID id.ModelID, _ *usecase.Oper
...
+	if err != nil {
+		return nil, err
+	}
+	s, err := i.repos.Schema.FindByID(ctx, g.Schema())
+	if err != nil {
+		return nil, err
+	}
+	return s, nil
+}
+
```
Comment (yk-eukarya) [inline server/internal/usecase/interactor/schema.go:86]:
```suggestion
```
you can use the schema package

File: server/internal/adapter/publicapi/schema_export.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+		fieldSchema := map[string]interface{}{
+			"type":        fieldType,
+			"title":       field.Name(),
+			"description": field.Description(),
+		}
+		if format != "" {
+			fieldSchema["format"] = format
+		}
+
+		var maxLength *int
```
Comment (yk-eukarya) [inline server/internal/adapter/publicapi/schema_export.go:45]:
```suggestion
```

**Review by yk-eukarya — COMMENTED** [review]
the duplicated logic between public api and integration api can be moved to `pkg/exporters`.
its better to be typed instead to define it as `map[string]any`.

File: server/internal/adapter/publicapi/schema_export.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+			Number: func(f *schema.FieldNumber) {
+				if min := f.Min(); min != nil {
+					fieldSchema["minimum"] = *min
+				}
+				if max := f.Max(); max != nil {
+					fieldSchema["maximum"] = *max
+				}
+			},
+			Group: func(f *schema.FieldGroup) {
+				gs, _ := uc.Schema.FindByGroup(ctx, f.Group(), nil)
```
[the remark this answers is not in the document]
Comment (nourbalaha) [reply server/internal/adapter/publicapi/schema_export.go:84]: These two endpoints lack access to the model ID, making it impossible to retrieve the schema package directly. How can I obtain the schema by group in such cases?
`/schemata/{schemaId}/schema.json`
`/projects/{projectIdOrAlias}/schemata/{schemaId}/schema.json`

**Review by yk-eukarya — APPROVED** [review]

**Pull request merged**